### ISSUE TITLE
all found build instructions seems to be out-of-date

### ISSUE THREAD

**Issue by msoulier** [resiprocate/resiprocate#303]
I looked here, I looked https://www.resiprocate.org/DeveloperQuickstart, and there is no configure.ac or Makefile.am file in the repo. The cmake instructions say that it is only used to build the tests, which does not appear to be the case. 

How should one build this project? Are there any up-to-date instructions?

The quickstart website also said to use the debian packages but there are none. 

I'm trying cmake but I seem to be missing something that I can't find in the debian repos.

```
msoulier@cortado:...cking/resiprocate$ cmake .
-- Found BerkeleyDb: /usr/lib/x86_64-linux-gnu/libdb.so  
-- Checking for module 'libpcre'
--   Found libpcre, version 8.39
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find cajun (missing: CAJUN_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  build/cmake_modules/Findcajun.cmake:17 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:571 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/msoulier/tracking/resiprocate/CMakeFiles/CMakeOutput.log".
See also "/home/msoulier/tracking/resiprocate/CMakeFiles/CMakeError.log".
```

msoulier@cortado:~$ sudo apt-cache search cajun
msoulier@cortado:~$

**Comment by sgodin** [comment]
@dpocock has been working on CMake support in GIT main.  I'm not sure offhand if he kept the previous autotools build files in place or not.  

Perhaps someone that works more with resip/linux on a daily basis can chime in here, since it seems that Daniel hasn't been present in this project over the last month or so.

Note:  Cajun is a header file only project, and is present in resip's contrib folder, perhaps you can just set the CAJUN_INCLUDE_DIR to point to it.

Scott

**Comment by msoulier** [comment]
Yeah I tried that and while cmake was happy, the build failed as the header include path was not updated to include that directory. I think I'd have to hack the build and put cajun into the include path.

Right now I've rolled back to 1.12.0 and autotools is working fine. I'm not a fan of cmake but I don't know it well. I'll have to take a tutorial on it.

**Comment by sgodin** [comment]
Sounds like the Cmake stuff still needs some more attention.  Hopefully Daniel will return to chime in soon.  

I did find this discussion on the pull request.... not sure if helps at all.
https://github.com/resiprocate/resiprocate/pull/107

**Comment by pbalash0v** [comment]
@msoulier (as a workaround) A minimal build of reSIProcate (i.e. no repro and some other stuff) on Linux can be done with disabling some features via `ccmake` tool (in debian this is `cmake-curses-gui` package). Just set some options to OFF. For me it was like this:

```
 BERKELEYDB_INCLUDE_DIR           BERKELEYDB_INCLUDE_DIR-NOTFOUND
 BERKELEYDB_LIBRARY_CXX           BERKELEYDB_LIBRARY_CXX-NOTFOUND
 BERKELEYDB_LIBRARY_DB            BERKELEYDB_LIBRARY_DB-NOTFOUND
 BUILD_CLICKTOCALL                OFF
 BUILD_DSO_PLUGINS                ON
 BUILD_ICHAT_GW                   OFF
 BUILD_P2P                        OFF
 BUILD_PYTHON                     OFF
 BUILD_QPID_PROTON                OFF
 BUILD_RECON                      OFF
 BUILD_RECONSERVER                OFF
 BUILD_REND                       OFF
 BUILD_REPRO                      OFF
 BUILD_RETURN                     OFF
 BUILD_SHARED_LIBS                OFF
 BUILD_TELEPATHY_CM               OFF
 BUILD_TFM                        OFF
 CMAKE_BUILD_TYPE
 CMAKE_INSTALL_PREFIX             /usr/local
 DEFAULT_BRIDGE_MAX_IN_OUTPUTS    OFF
 ENABLE_ANDROID                   OFF
 ENABLE_LOG_REPOSITORY_DETAILS    OFF
 PEDANTIC_STACK                   OFF
 RECON_LOCAL_HW_TESTS             OFF
 REGENERATE_MEDIA_SAMPLES         ON
 RESIP_ASSERT_SYSLOG              ON
 RESIP_HAVE_RADCLI                OFF
 SOX_EXECUTABLE                   /usr/bin/sox
 USE_CONTRIB                      OFF
 USE_DTLS                         ON
 USE_FMT                          OFF
 USE_GSTREAMER                    OFF
 USE_IPV6                         ON
 USE_KURENTO                      OFF
 USE_LIBWEBRTC                    OFF
 USE_MAXMIND_GEOIP                OFF
 USE_MYSQL                        OFF
 USE_NDKPORTS_HACKS               OFF
 USE_NETSNMP                      OFF
 USE_POPT                         OFF
 USE_POSTGRESQL                   OFF
 USE_SIGCOMP                      OFF
 USE_SIPXTAPI                     OFF
 USE_SOCI_MYSQL                   OFF
 USE_SOCI_POSTGRESQL              OFF
 USE_SRTP1                        OFF
 VERSIONED_SONAME                 ON
 WITH_C_ARES                      OFF
 WITH_SSL                         ON
```

**Comment by fgw-auerswald** [comment]
I tried to build the current master branch undeer linux and had the same results. 
- None of the instructions worked
- some where impossible to follow
- input files for configure / autoreconf are missing
- cmake configs are somewhat brocken as well
- links to download source releases (which are supposed to have the cofigure files) are all brocken
- releases for most OS specific repos were dropped/are not available for current versions of the OS anymore
Is anyone using Linux still actively working on this or is everyone just trying to keep their existing branches alive?

**Comment by alienpenguin** [comment]
> I tried to build the current master branch undeer linux and had the same results.
> 
> * None of the instructions worked
> * some where impossible to follow
> * input files for configure / autoreconf are missing
> * cmake configs are somewhat brocken as well
> * links to download source releases (which are supposed to have the cofigure files) are all brocken
> * releases for most OS specific repos were dropped/are not available for current versions of the OS anymore
> 
> Is anyone using Linux still actively working on this or is everyone just trying to keep their existing branches alive?

Hi, i was able to compile the stack on linux using cmake (i did disable all the other projects because i was not interested in compiling / fixing compilation for them) moslty following the cmake configuration above. If you need specific info let me know
